pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/generator/EnvironmentDirective/config.jelly

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ The MIT License
+  ~
+  ~ Copyright (c) 2018, CloudBees, Inc.
+  ~
+  ~ Permission is hereby granted, free of charge, to any person obtaining a copy
+  ~ of this software and associated documentation files (the "Software"), to deal
+  ~ in the Software without restriction, including without limitation the rights
+  ~ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  ~ copies of the Software, and to permit persons to whom the Software is
+  ~ furnished to do so, subject to the following conditions:
+  ~
+  ~ The above copyright notice and this permission notice shall be included in
+  ~ all copies or substantial portions of the Software.
+  ~
+  ~ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  ~ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  ~ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  ~ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  ~ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  ~ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  ~ THE SOFTWARE.
+  -->
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:f="/lib/form" xmlns:l="/lib/layout" xmlns:c="/lib/credentials" xmlns:st="jelly:stapler">
+    <f:block>
+        See <a href="https://jenkins.io/doc/book/pipeline/syntax/#environment" target="_blank">the online documentation</a>
+        for more information on the <code>environment</code> directive.
+        <f:entry title="${%Environment Variables}" help="${descriptor.getHelpPage()}">
+            <f:repeatable var="env" items="${instance.getEnv()}">
+                <table width="100%">
+                    <f:entry title="${%Variable Name}" help="${descriptor.getEnvHelp('name')}">
+                        <f:textbox name="env.name" value="${env.name}" />
+                    </f:entry>
+                    <!-- TODO: Figure out how to support credentials here. -->
```

Comment:
Two different add buttons? Or a dropdown add button maybe.

Reply:
Yeah, the problems I had with credentials were...weird. Something to revisit in the future.